pkg/sql/logictest/testdata/logic_test/alter_column_type

```diff
@@ -442,3 +435,39 @@ CREATE TABLE t22 (x INT);
 
 statement error pq: unimplemented: ALTER COLUMN TYPE cannot be used in combination with other ALTER TABLE commands
 ALTER TABLE t22 ALTER COLUMN x SET NOT NULL, ALTER COLUMN x TYPE STRING;
+
+# Ensure ALTER COLUMN TYPE USING EXPRESSION works.
+statement ok
+CREATE TABLE t23 (x INT);
+INSERT INTO t23 VALUES (-3), (-2), (-1), (0), (1), (2), (3)
+
+statement ok
+ALTER TABLE t23 ALTER COLUMN x TYPE BOOL USING (x > 0)
```

Comment:
What happens if the using expression doesn't agree with the new type of the column? What if there are invalid variable references? Are there certain classes of functions that we allow in the using expression? It would probably be similar to what we allow in computed columns.

Reply:
We do that to disable invalid variable references but I'm not sure if we would have the same set of disallowed functions as computed columns. Postgres seems to allow impure functions for the expression - not sure if we would want to support that.